inference-engine/tests/functional/plugin/cpu/single_layer_tests/roi_pooling.cpp

```diff
@@ -0,0 +1,210 @@
+                }
+            } else {
+                blob = GenerateInput(*info);
+            }
+            inputs.push_back(blob);
+            it++;
+        }
+    }
+
+    void SetUp() {
+        LayerTestsDefinitions::roiPoolingParamsTuple basicParamsSet;
+        CPUSpecificParams cpuParams;
+        std::map<std::string, std::string> additionalConfig;
+
+        InferenceEngine::SizeVector inputShape;
+        InferenceEngine::SizeVector coordsShape;
+        InferenceEngine::SizeVector poolShape;
+        InferenceEngine::Precision netPrecision;
+
+        std::tie(basicParamsSet, cpuParams, additionalConfig) = this->GetParam();
+        std::tie(inFmts, outFmts, priority, selectedType) = cpuParams;
+        std::tie(inputShape, coordsShape, poolShape, spatial_scale, pool_method, netPrecision, targetDevice) = basicParamsSet;
+
+        if (additionalConfig[PluginConfigParams::KEY_ENFORCE_BF16] == PluginConfigParams::YES)
+            inPrc = outPrc = netPrecision = Precision::BF16;
+        else
+            inPrc = outPrc = netPrecision;
+
+        auto ngPrc = FuncTestUtils::PrecisionUtils::convertIE2nGraphPrc(netPrecision);
+        auto params = ngraph::builder::makeParams(ngPrc, {inputShape, coordsShape});
+        auto paramOuts = ngraph::helpers::convert2OutputVector(ngraph::helpers::castOps2Nodes<ngraph::op::Parameter>(params));
+
+        std::shared_ptr<ngraph::Node> roi_pooling = ngraph::builder::makeROIPooling(paramOuts[0], paramOuts[1], poolShape, spatial_scale, pool_method);
+        ngraph::ResultVector results{std::make_shared<ngraph::opset3::Result>(roi_pooling)};
+
+        function = makeNgraphFunction(ngPrc, params, roi_pooling, "roi_pooling");
+
+        selectedType += "_";
+        selectedType += netPrecision.name();
+    }
+
+private:
+    ngraph::helpers::ROIPoolingTypes pool_method;
+    float spatial_scale;
+};
+
+TEST_P(ROIPoolingCPULayerTest, CompareWithRefs) {
+    SKIP_IF_CURRENT_TEST_IS_DISABLED()
+    Run();
+    CheckPluginRelatedResults(executableNetwork, "ROIPooling");
+}
+
+namespace {
+
+std::vector<std::map<std::string, std::string>> additionalConfig
+    = {{{PluginConfigParams::KEY_ENFORCE_BF16, PluginConfigParams::NO}},
+       {{PluginConfigParams::KEY_ENFORCE_BF16, PluginConfigParams::YES}}};
+
+std::vector<CPUSpecificParams> filterCPUInfoForDevice() {
+    std::vector<CPUSpecificParams> resCPUParams;
+    if (with_cpu_x86_avx512f()) {
+        resCPUParams.push_back(CPUSpecificParams{{nChw16c, nc}, {nChw16c}, {"jit_avx512"}, "jit_avx512"});
+    } else if (with_cpu_x86_avx2()) {
+        resCPUParams.push_back(CPUSpecificParams{{nChw8c, nc}, {nChw8c}, {"jit_avx2"}, "jit_avx2"});
+    } else if (with_cpu_x86_sse42()) {
+        resCPUParams.push_back(CPUSpecificParams{{nChw8c, nc}, {nChw8c}, {"jit_sse42"}, "jit_sse42"});
+    } else {
+        resCPUParams.push_back(CPUSpecificParams{{nChw8c, nc}, {nChw8c}, {"ref"}, "ref"});
+    }
+
+    return resCPUParams;
+}
```

Comment:
It is better to use the `filterCPUInfoForDevice(std::vector<CPUSpecificParams> CPUParams)` from the test utils. Just make the configurations vector here and pass it to the `filterCPUInfoForDevice` in every test instance.

Reply:
Discussed.
Updated